packages/wonder-blocks-modal/util/get-nearest-modal-launcher-portal.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+import {ModalLauncherPortalAttributeName} from "./constants.js";
+
+/**
+ * Finds the nearest modal launcher portal element in which a given element
+ * is nested.
+ * @param {null | Element | Text} element The element from which to find the nearest
+ * portal
+ * @returns {Element | null} The nearest parent modal launcher portal or
+ * `null`.
+ */
+export default function getNearestModalLauncherPortal(
+    element: null | Element | Text,
+) {
+    let candidateElement = element ? element.parentElement : null;
+    while (
+        candidateElement &&
+        !candidateElement.hasAttribute(ModalLauncherPortalAttributeName)
+    ) {
+        if (candidateElement === candidateElement.parentElement) {
```

Comment:
This probably isn't needed. IIRC, some browsers would return `document.body` as its own parent, so I wanted to make sure we terminated in that case.